src/client/ClientGameRunner.ts

```diff
@@ -187,9 +187,10 @@ export class ClientGameRunner {
   }
 
   private saveGame(update: WinUpdate) {
+    if (this.myPlayer === null) throw new Error("Not initialized");
     const players: PlayerRecord[] = [
       {
-        ip: null,
+        playerID: this.myPlayer.id(),
```

Comment:
Instead of throwing this could be:
```suggestion
        playerID: this.myPlayer?.id() ?? "",
```